plugins/experimental/slice/util.cc

```diff
@@ -86,7 +86,8 @@ request_block(TSCont contp, Data *const data)
   }
 
   // create virtual connection back into ATS
-  TSVConn const upvc = TSHttpConnectWithPluginId(reinterpret_cast<sockaddr *>(&data->m_client_ip), PLUGIN_NAME, 0);
+  TSVConn const upvc = TSHttpConnectPlugin(reinterpret_cast<sockaddr *>(&data->m_client_ip), PLUGIN_NAME, 0, data->m_buffer_index,
```

Comment:
Why drop the plugin ID? That's quite useful.

Reply:
I didn't, it's still there. See the corresponding API change in PR #8088 [here](https://github.com/apache/trafficserver/pull/8088/files#diff-e188d413739d07f1ad183ae5912ca435557809252154bdd5f2fcf625f95b9600R6882) that adds this new function signature.

The problem is I didn't want to keep adding to the function name, so I chopped that part off when I added two new arguments. Otherwise I would have needed to add `andBufferIndexAndBufferWaterMark` or similar to the new function and that seemed excessive when the arguments are self-explanatory. The `TSHttpConnectWithPluginId` function still exists and the arguments above are identical to what they were previously, we just call the less-specifically-named function that has more arguments here to provide the buffer index and watermark.